nilearn/tests/test_datasets.py

```diff
@@ -27,14 +27,22 @@
 file_mock = None
 
 
+def setup_silence():
+    # Wrap functions with versions that have verbose=0
+    datasets._get_dataset_dir = silence_fn(datasets._get_dataset_dir)
+    datasets._uncompress_file = silence_fn(datasets._uncompress_file)
+    datasets._fetch_file = silence_fn(datasets._fetch_file)
+    datasets._fetch_files = silence_fn(datasets._fetch_files)
+    datasets._chunk_read_ = silence_fn(datasets._chunk_read_)
```

Comment:
On further thought, I think the *best* way to do this is to automatically silence all functions--there should be no manual verification in tests, so output simply isn't helpful.  To test verbosity of any function, people should write specific unit tests.

What I've done is shorthand for that.  I could use the `inspect` function to recursively search modules for functions that accept a `verbose` parameter, and I could write a teardown function as well.

This seems the most complete and safest to me.  It could also wrap `stderr` / `stdout` to test whether any function has output... in fact, doing this for the entire test suite via a `TestRunner`, rather than using a decorator, is also an option.